Source/CombatExtended/CombatExtended/Comps/CompAmmosetSwitcher.cs

```diff
@@ -0,0 +1,262 @@
+
+                if (compAmmo.Wielder.jobs?.curJob?.def == CE_JobDefOf.ReloadWeapon)
+                {
+                    compAmmo.Wielder.jobs.EndCurrentJob(Verse.AI.JobCondition.InterruptForced);
+                }
+            }
+            compEq.PrimaryVerb.verbProps.burstShotCount = this.Props.propsFireModesUnderBarrel.aimedBurstShotCount;
+            usingUnderBarrel = true;
+        }
+
+        public void SwithToB()
+        {
+            UnderBarrelLoadedAmmo = compAmmo.CurrentAmmo;
+            UnderBarrelMagCount = compAmmo.CurMagCount;
+
+            compAmmo.props = compPropsAmmo;
+            compEq.PrimaryVerb.verbProps = defVerbProps.MemberwiseClone();
+            compFireModes.props = compPropsFireModes;
+            compAmmo.CurMagCount = mainGunMagCount;
+            compAmmo.CurrentAmmo = mainGunLoadedAmmo;
+            compAmmo.SelectedAmmo = compAmmo.CurrentAmmo;
+            if (compAmmo.Wielder != null)
+            {
+                compAmmo.Wielder.TryGetComp<CompInventory>().UpdateInventory();
+
+                if (compAmmo.Wielder.jobs?.curJob?.def == CE_JobDefOf.ReloadWeapon)
+                {
+                    compAmmo.Wielder.jobs.EndCurrentJob(Verse.AI.JobCondition.InterruptForced);
+                }
+            }
+            compEq.PrimaryVerb.verbProps.burstShotCount = defVerbProps.burstShotCount;
+            usingUnderBarrel = false;
+        }
+
+        public override IEnumerable<Gizmo> CompGetGizmosExtra()
+        {
+            if (compAmmo.Props.ammoSet == compPropsAmmo.ammoSet)
+            {
+                yield return new Command_Action
+                {
+
+                    defaultLabel = "CE_SwitchAmmmoSetToUnderBarrel".Translate(),
+                    icon = ContentFinder<Texture2D>.Get("UI/Buttons/Reload"),
+                    defaultDesc = "CE_UBGLStats".Translate() +
+                    "\n " + "WarmupTime".Translate() + ": " + Props.verbPropsUnderBarrel.warmupTime
+                    + "\n " + "Range".Translate() + ": " + Props.verbPropsUnderBarrel.range
+                    + "\n " + "CE_AmmoSet".Translate() + ": " + Props.propsUnderBarrel.ammoSet.label
+                    + "\n " + "CE_MagazineSize".Translate() + ": " + Props.propsUnderBarrel.magazineSize
+                    ,
+                    action = delegate
+                    {
+                        SwitchToUB();
+                    }
+                };
+            }
+            else
+            {
+                yield return new Command_Action
+                {
+
+                    defaultLabel = "CE_SwitchAmmmoSetToNormalRifle".Translate(),
+                    icon = ContentFinder<Texture2D>.Get("UI/Buttons/Reload"),
+                    action = delegate
+                    {
+                        SwithToB();
+                    }
+                };
+            }
+        }
+
+        public override string TransformLabel(string label)
+        {

```

Comment:
If the commented out section is really not needed, the whole function can go away, unless you need it for a Harmony target.